### PR TITLE
Fix timing issue with ProcessPendingContributions browser test

### PR DESCRIPTION
**Pull request by emerick**
Resolves https://github.com/brave/brave-browser/issues/9687

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [x] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

Run the following command and verify that the test passes:

```
brave_browser_tests --gtest_filter=BraveRewardsBrowserTest.ProcessPendingContributions
```

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_rewards/browser/rewards_service_browsertest.cc [hunk 1]
```diff
@@ -1234,7 +1245,7 @@ class BraveRewardsBrowserTest
       multiple_tip_reconcile_count_++;
       multiple_tip_reconcile_status_ = converted_result;
 
-      if (multiple_tip_reconcile_count_ == multiple_tip_reconcile_needed_) {
+      if (multiple_tip_reconcile_count_ >= multiple_tip_reconcile_needed_) {
```
Comment (NejcZdovc) [inline components/brave_rewards/browser/rewards_service_browsertest.cc:1248]: why was this change needed?

File: components/brave_rewards/browser/rewards_service_browsertest.cc [hunk 2]
```diff
@@ -945,9 +945,20 @@ class BraveRewardsBrowserTest
   }
 
   void RefreshPublisherListUsingRewardsPopup() const {
-    rewards_service_browsertest_utils::WaitForElementThenClick(
-        OpenRewardsPopup(),
-        "[data-test-id='unverified-check-button']");
+    // Refresh publisher list; assume that absence of refresh button
```
Comment (NejcZdovc) [inline components/brave_rewards/browser/rewards_service_browsertest.cc:948]: how come publisher is already verified if we didn't click refresh yet?

Comment (emerick) [reply components/brave_rewards/browser/rewards_service_browsertest.cc:948]: From my understanding of the code the publisher list also updates on a timer, so sometimes the manual click isn't needed.

File: components/brave_rewards/browser/rewards_service_browsertest.cc (hunk 1 above)
Comment (emerick) [reply components/brave_rewards/browser/rewards_service_browsertest.cc:1248]: Yeah, that isn't needed. I had a faulty understanding of what was going on there. Now I see that the issue isn't here. The problem is that we can hit the required number of reconciliations before even calling the function to wait for them. Fixed now.

File: components/brave_rewards/browser/rewards_service_browsertest.cc (hunk 2 above)
Comment (NejcZdovc) [reply components/brave_rewards/browser/rewards_service_browsertest.cc:955]: interesting so you thing that timer is causing problems

Comment (emerick) [reply components/brave_rewards/browser/rewards_service_browsertest.cc:955]: Yeah, the timer is kicking off the update and updating the server before there's any chance to click the button (at least on my local Windows machine). Once the rewards popup opens, there's no longer a "Refresh Status" button to click on because it's already updated to a verified publisher.

Comment (NejcZdovc) [reply components/brave_rewards/browser/rewards_service_browsertest.cc:955]: this happened because we don't send any end when doing pagination, so pagination will continue for 100 pages and because of that we have this race condition

**Comment by emerick** [comment]
@NejcZdovc found the real fix here https://github.com/brave/brave-core/pull/5458, closing.
